plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/editor.scss

```
@@ -96,3 +96,13 @@ $max-button-width: calc((100% - #{ $total-gap-width }) / #{ $max-columns });
 		text-align: end;
 	}
 }
+
+.editor-styles-wrapper .wp-block-woocommerce-product-collection .wp-block-post-title {
+	a:where(:not(.wp-element-button)) {
+		&:hover {
+			text-decoration: none;
+		}
+
+		cursor: default;
+	}
```

There's still a color change happening on hover. Also, I think it's fine to keep the decoration to highlight it's going to be a link. It's the case for other links wrapped by the `Disabled` component so we should follow the same.

Hence my suggestion would be to replace the above with:

```suggestion
	a:where(:not(.wp-element-button)) {
		pointer-events: none
	}
```

`pointer-events: none` takes care of interactions in general and doesn't react on hover or click which is more generic (and simpler) solution.